### PR TITLE
add a search page, add search to the 404 page, and an autocomplete search to all pages.

### PR DESCRIPTION
**Pull request by SvenDowideit**

### CODE REVIEW

File: content/search.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
++++
+title = "Search the Docker Documentation"
+description = "Docker documentation Search"
+keywords = ["Search, Docker, documentation, manual, guide, reference, api"]
+notoc = true
++++
+
+# Search the Docker Documentation
+
+<input type="text" id="search-box" name="q" />
```
Comment (thaJeztah) [inline content/search.md]: perhaps `type="search"`, or does that mess up styling?

File: content/search.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+      <div class="ds-suggestion">
+        <div class="algolia-docsearch-suggestion algolia-docsearch-suggestion__main algolia-docsearch-suggestion__secondary " style="white-space: normal;" id="hits-container"></div>
+      </div>
+    </span>
+  </div> </div> </span>
+<div id="pagination-container"></div>
+
+
+<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/instantsearch.js/1/instantsearch.min.css">
+<link rel="stylesheet" type="text/css" href="//cdn.jsdelivr.net/docsearch.js/2/docsearch.min.css">
```
Comment (thaJeztah) [inline content/search.md]: should we add these to the `<head>`? May result in better performance

(and the scripts below to the bottom of the page)

File: content/search.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
++++
+title = "Search the Docker Documentation"
+description = "Docker documentation Search"
+keywords = ["Search, Docker, documentation, manual, guide, reference, api"]
+notoc = true
```
Comment (sanscontext) [inline content/search.md]: OMG! I didn't know about this flag. Totally going to use that.

File: content/search.md (hunk 1 above)
Comment (SvenDowideit) [inline content/search.md]: yeah, it'll change it - on _some_ browsers

File: content/search.md (hunk 2 above)
Comment (SvenDowideit) [inline content/search.md]: They're in the md file so they don't get loaded for all pages - I'll look into how to extract into a conditional in a later PR

good point tho.

File: content/search.md (hunk 3 above)
Comment (SvenDowideit) [inline content/search.md]: DOH :) who wrote the docs for this thing.

**Comment by sanscontext** [comment]
LGTM, 🚀

**Pull request merged**